matrix/max_area_of_island.py

```diff
@@ -0,0 +1,134 @@
+# Given an m x n binary matrix grid. An island is a group of 1's
+# (representing land) connected 4-directionally (horizontal or vertical.)
+# You may assume all four edges of the grid are surrounded by water.
+# The area of an island is the number of cells with a value 1 in the island.
+# Return the maximum area of an island in a grid.
+# If there is no island, return 0.
+
+
+def is_safe(row: int, col: int, rows: int, cols: int) -> bool:
+    """
+    Checking weather coordinate (row,col) is valid or not.
+
+    >>> is_safe(0, 0, 5, 5)
+    True
+
+    >>> is_safe(-1,-1, 5, 5)
+    False
+    """
+    return not (row < 0 or col < 0 or row >= rows or col >= cols)
+
+
+def dfs(
+    row: int, col: int, rows: int, cols: int, seen: set, mat: list[list[int]]
+) -> int:
+    """
+    Returns the current area of the island
+    >>> dfs(0,0, 8,8,set(), [ \
```

Comment:
```suggestion
    >>> dfs(0,0, 8,8,set(), [
```
The `\` is not necessary

Reply:
removed `\`